stacks/end-to-end-security/krb5.yaml

```diff
@@ -14,7 +14,7 @@ spec:
     spec:
       initContainers:
         - name: init
-          image: docker.stackable.tech/stackable/krb5:1.18.2-stackable24.7.0 # TODO: bump to 1.21.1?
+          image: docker.stackable.tech/stackable/krb5:1.21.1-stackable24.7.0
```

Comment:
If this PR is going to main, then I believe all of these images should be 0.0.0-dev (in preparation for the next release).

In future, the release scripts check for 0.0.0-dev and bump them to the release at the time for the release branch.

So I think either:
- Change the base branch for the PR to `release-24.7`, or
- Change this to `0.0.0-dev`.

Perhaps I miss something about how we do demos, but this is my current understanding.

Reply:
We always point to the latest release (24.7 now) for stability.

Reply:
Ok cool. Maybe we can change that once we properly support release branches?

Will approve